pkg/storage/storagewrappers/sharediterator/shared_iterator_datastore.go

```diff
@@ -43,10 +44,13 @@ var (
 		Name:      "shared_iterator_bypassed",
 		Help:      "Total number of iterators bypassed by the shared iterator layer because the internal map size exceed specified limit.",
 	}, []string{"operation"})
+
+	errSharedIteratorWatchdog = errors.New("shared iterator watchdog timeout")
 )
 
 const (
-	defaultSharedIteratorLimit = 1000000
+	defaultSharedIteratorLimit        = 1000000
+	defaultSharedIteratorMaxAliveTime = 4 * time.Minute
```

Comment:
Why 4 mins?

Reply:
My worry is that this will have a goroutine (per each of the entries of the map rather than 1 since my last review) for up to 4 mins. Multiplying that by the size and a potential leak and that would be quite bad, no?

Reply:
Reduced to 1 minute + request timeout